HealthVault/Classes/Caching/Private/MHVThingCacheDatabase.m

```diff
@@ -759,6 +778,231 @@ - (void)updateLastCompletedSyncDate:(NSDate *_Nullable)lastCompletedSyncDate
      }];
 }
 
+- (void)cachePendingMethod:(MHVPendingMethod *)pendingMethod
+                completion:(void (^)(NSError *_Nullable error))completion
+{
+    MHVASSERT_PARAMETER(pendingMethod);
+    
+    if (!pendingMethod)
+    {
+        if (completion)
+        {
+            completion([NSError MVHRequiredParameterIsNil]);
+        }
+        
+        return;
+    }
+    
+    if (!self.isDatabaseReady)
+    {
+        if (completion)
+        {
+            completion([NSError MHVCacheDeleted]);
+        }
+        return;
+    }
+    
+    [self.managedObjectContext performBlock:^
+    {
+        MHVCachedRecord *record = (MHVCachedRecord *)[self fetchCachedRecord:pendingMethod.recordId.UUIDString];
+        
+        if (!record)
+        {
+            if (completion)
+            {
+                completion([NSError MHVCacheError:@"Record could not be found"]);
+            }
+            
+            return;
+        }
+        
+        // Support for updating a pending method request - If a pending method with the same identifier exists update rather
+        // than creating a new one.
+        MHVPendingThingOperation *operation = [record pendingThingOpertaionWithIdentifier:pendingMethod.identifier];
+        
+        if (!operation)
+        {
+            operation = [self newPendingThingOperationForRecord:record];
+            
+            if (!operation)
+            {
+                if (completion)
+                {
+                    completion([NSError MHVCacheError:@"Could not create a new MHVPendingThingOperation."]);
+                }
+                return;
+            }
+            
+        }
+        
+        operation.name = pendingMethod.name;
+        operation.version = pendingMethod.version;
+        operation.originalRequestDate = pendingMethod.originalRequestDate;
+        operation.parameters = pendingMethod.parameters;
+        operation.identifier = pendingMethod.identifier;
+        operation.correlationId = pendingMethod.correlationId.UUIDString;
```

Comment:
I actually think the lowercase string is a bad idea and will be filing a bug to fix it later. Whenever possible, we should be passing NSUUIDs instead of strings. NSUUID.UUIDstring returns a capitalized string - If anything we should be converting to capitalized strings - But in reality we should keep the NSUUID until it is stored in the database.